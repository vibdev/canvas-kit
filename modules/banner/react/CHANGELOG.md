# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.25](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.24...@workday/canvas-kit-react-banner@0.1.25) (2019-06-20)


### Bug Fixes

* broad stroke accessibility fixes ([9803412](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/commits/9803412))





## [0.1.24](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.23...@workday/canvas-kit-react-banner@0.1.24) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.23](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.22...@workday/canvas-kit-react-banner@0.1.23) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.22](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.21...@workday/canvas-kit-react-banner@0.1.22) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.21](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.20...@workday/canvas-kit-react-banner@0.1.21) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.20](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.19...@workday/canvas-kit-react-banner@0.1.20) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.19](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.18...@workday/canvas-kit-react-banner@0.1.19) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.18](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.17...@workday/canvas-kit-react-banner@0.1.18) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.16...@workday/canvas-kit-react-banner@0.1.17) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.15...@workday/canvas-kit-react-banner@0.1.16) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.14...@workday/canvas-kit-react-banner@0.1.15) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.13...@workday/canvas-kit-react-banner@0.1.14) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.12...@workday/canvas-kit-react-banner@0.1.13) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.11...@workday/canvas-kit-react-banner@0.1.12) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.10...@workday/canvas-kit-react-banner@0.1.11) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.9...@workday/canvas-kit-react-banner@0.1.10) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.8...@workday/canvas-kit-react-banner@0.1.9) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.7...@workday/canvas-kit-react-banner@0.1.8) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.6...@workday/canvas-kit-react-banner@0.1.7) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.5...@workday/canvas-kit-react-banner@0.1.6) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.4...@workday/canvas-kit-react-banner@0.1.5) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.3...@workday/canvas-kit-react-banner@0.1.4) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.2...@workday/canvas-kit-react-banner@0.1.3) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.1...@workday/canvas-kit-react-banner@0.1.2) (2019-05-06)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## 0.1.1 (2019-04-30)

**Note:** Version bump only for package @workday/canvas-kit-react-banner
