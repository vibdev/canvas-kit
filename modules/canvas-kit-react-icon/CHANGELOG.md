# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.3.3"></a>
## [0.3.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/compare/@workday/canvas-kit-react-icon@0.3.2...@workday/canvas-kit-react-icon@0.3.3) (2018-04-24)




**Note:** Version bump only for package @workday/canvas-kit-react-icon

<a name="0.3.2"></a>
## [0.3.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/compare/@workday/canvas-kit-react-icon@0.3.1...@workday/canvas-kit-react-icon@0.3.2) (2018-04-17)




**Note:** Version bump only for package @workday/canvas-kit-react-icon

<a name="0.3.1"></a>
## [0.3.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/compare/@workday/canvas-kit-react-icon@0.3.0...@workday/canvas-kit-react-icon@0.3.1) (2018-04-13)




**Note:** Version bump only for package @workday/canvas-kit-react-icon

<a name="0.3.0"></a>
# [0.3.0](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/compare/@workday/canvas-kit-react-icon@0.2.3...@workday/canvas-kit-react-icon@0.3.0) (2018-04-13)


### Features

* **react-16:** support React 16 peerDependency ([003a78e](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/003a78e))




<a name="0.2.3"></a>
## [0.2.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/compare/@workday/canvas-kit-react-icon@0.2.2...@workday/canvas-kit-react-icon@0.2.3) (2018-04-12)


### Bug Fixes

* **icon:** pass props to Icon component ([680c705](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/680c705))
* **icon:** use name prop ([281f938](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/281f938))




<a name="0.2.2"></a>
## [0.2.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/compare/@workday/canvas-kit-react-icon@0.2.1...@workday/canvas-kit-react-icon@0.2.2) (2018-03-19)




**Note:** Version bump only for package @workday/canvas-kit-react-icon

<a name="0.2.1"></a>
## [0.2.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/compare/@workday/canvas-kit-react-icon@0.2.0...@workday/canvas-kit-react-icon@0.2.1) (2018-03-13)


### Bug Fixes

* **dependencies:** add missing dependencies and peerDeps ([7201f93](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/7201f93))




<a name="0.2.0"></a>
# 0.2.0 (2018-02-20)


### Features

* **icon:** add accent icon color styles ([0cfabbb](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/0cfabbb))
* **icon:** add applet coloring ([b74e862](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/b74e862))
* **icon:** add configurable color properties ([0d52453](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/0d52453))
* **icon:** add default system icon coloring ([c3fb22f](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/c3fb22f))
* **icon:** add icon component ([75cc0b8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/75cc0b8))
* **icon:** add size prop ([d14d9da](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/d14d9da))
* **icon:** add transparent prop ([6b0912a](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/6b0912a))
* **icon:** initialize module ([e029082](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon/commits/e029082))
