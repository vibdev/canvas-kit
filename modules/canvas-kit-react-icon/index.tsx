import React, { Component } from 'react'
import canvasAccentIconsWeb, {
  CanvasAccentIcons as AccentIcons,
} from '@workday/canvas-accent-icons-web'
import canvasAppletIconsWeb, {
  CanvasAppletIcons as AppletIcons,
} from '@workday/canvas-applet-icons-web'
import canvasSystemIconsWeb, {
  CanvasSystemIcons as SystemIcons,
} from '@workday/canvas-system-icons-web'

export enum IconSets {
  Accent,
  Applet,
  System,
}

export { AccentIcons, AppletIcons, SystemIcons }

type Props = {
  name: string
  set?: IconSets
}

export default class Icon extends Component<Props, {}> {
  public render() {
    let iconSet

    switch (this.props.set) {
      case IconSets.Accent:
        iconSet = canvasAccentIconsWeb
        break
      case IconSets.Applet:
        iconSet = canvasAppletIconsWeb
        break
      case IconSets.System:
      default:
        iconSet = canvasSystemIconsWeb
    }

    if (!(this.props.name in iconSet)) {
      throw Error(`Icon "${this.props.name}" not found in "${IconSets[this.props.set]}" icon set`)
    }

    const icon = iconSet[this.props.name]

    return <span dangerouslySetInnerHTML={{ __html: icon }} {...this.props} />
  }
}
