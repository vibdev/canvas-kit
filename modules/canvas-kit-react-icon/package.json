{
  "name": "@workday/canvas-kit-react-icon",
  "version": "1.5.14",
  "description": "Canvas icons",
  "homepage": "https://workdaydesign.com",
  "author": "Joseph Le <joseph.le@workday.com>",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-icon"
  },
  "files": [
    "dist/",
    "lib/",
    "index.tsx"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean --parallel build:cjs build:es6",
    "build": "npm-run-all --parallel build:cjs build:es6"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "icon"
  ],
  "peerDependencies": {
    "react": ">= 15 < 17"
  },
  "dependencies": {
    "@workday/canvas-kit-react-core": "^1.9.6",
    "@workday/design-assets-types": "^0.2.3",
    "emotion": "^9.2.6",
    "react-emotion": "^9.2.6"
  },
  "devDependencies": {
    "@workday/canvas-accent-icons-web": "^1.0.0",
    "@workday/canvas-applet-icons-web": "^0.17.50",
    "@workday/canvas-system-icons-web": "^1.0.1"
  }
}
