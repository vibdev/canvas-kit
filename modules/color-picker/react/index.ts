import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
