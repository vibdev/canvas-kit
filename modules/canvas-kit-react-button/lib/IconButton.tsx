import * as React from 'react';
import {getButtonStyle} from './ButtonBase';
import styled from 'react-emotion';
import {IconButtonTypes} from './types';
import {BaseButtonProps} from './Button';
import {iconButtonStyles} from './ButtonStyles';
import {colors} from '@workday/canvas-kit-react-core';

export interface IconButtonProps extends BaseButtonProps<IconButtonTypes> {
  toggled?: boolean;
}

const IconButtonCon = styled('button')<IconButtonProps>(
  iconButtonStyles.styles,
  ({buttonType}) => getButtonStyle(iconButtonStyles, buttonType),
  ({buttonType, toggled}) => {
    if (!toggled) {
      return {};
    }

    switch (buttonType) {
      default:
        return {};
      case IconButtonTypes.Default:
        return {
          '&:focus&:hover, &:focus': {
            backgroundColor: colors.blueberry500,
          },
          '&:hover': {
            backgroundColor: colors.blueberry400,
          },
          backgroundColor: colors.blueberry500,
          borderColor: colors.blueberry500,
          ...getFillSelector(colors.frenchVanilla100),
          ...getAccentSelector(colors.frenchVanilla100),
        };
      case IconButtonTypes.IconPrimary:
        return {
          ...getBackgroundSelector(colors.blueberry400),
          ...getFillSelector(colors.blueberry400),
          ...getAccentSelector(colors.frenchVanilla100),
        };
      case IconButtonTypes.IconPrimaryFilled:
        return {
          backgroundColor: colors.blueberry400,
          '&:focus, &:hover': {
            backgroundColor: colors.blueberry500,
          },
          ...getBackgroundSelector(colors.frenchVanilla100),
          ...getFillSelector(colors.frenchVanilla100),
          ...getAccentSelector(colors.blueberry400),
        };
      case IconButtonTypes.IconInverse:
        return {
          '&:hover span .wd-icon-fill, span .wd-icon-fill': {
            fill: colors.frenchVanilla100,
          },
          ...getBackgroundSelector(colors.frenchVanilla100),
          ...getAccentSelector(colors.licorice200),
        };
      case IconButtonTypes.IconInverseFilled:
        return {
          backgroundColor: 'rgba(0, 0, 0, 0.4)',
          '&:focus': {
            backgroundColor: 'rgba(0, 0, 0, 0.4)',
          },
          ...getBackgroundSelector(colors.frenchVanilla100),
          ...getFillSelector(colors.frenchVanilla100),
          ...getAccentSelector(colors.licorice200),
        };
    }
  }
);

function getFillSelector(fillColor: string) {
  return {
    '&:focus span .wd-icon-fill, &:hover span .wd-icon-fill, span .wd-icon-fill': {
      fill: fillColor,
    },
  };
}

function getBackgroundSelector(fillColor: string) {
  return {
    '&:hover span .wd-icon-background, span .wd-icon-background': {
      fill: fillColor,
    },
  };
}

function getAccentSelector(fillColor: string) {
  return {
    '&:focus span .wd-icon-accent, &:hover span .wd-icon-accent, span .wd-icon-accent': {
      fill: fillColor,
    },
  };
}

export default class IconButton extends React.Component<IconButtonProps> {
  public static Types = IconButtonTypes;

  static defaultProps = {
    buttonType: IconButtonTypes.Default,
  };

  public render() {
    return (
      // TODO (breaking change): need to remove buttonType and buttonSize prop here, doesn't make sense to expose
      <IconButtonCon {...this.props} buttonSize={undefined} aria-pressed={this.props.toggled} />
    );
  }
}
