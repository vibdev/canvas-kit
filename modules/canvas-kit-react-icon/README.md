# Canvas Kit Icon

Components implementing Canvas accent, applet, and system icons.

## Usage

### Accent Icons

```jsx
import { colors } from '@workday/canvas-kit-react-core'
import { AccentIcon } from '@workday/canvas-kit-react-icon'

<AccentIcon name={AccentIcon.Icons.Shield} />
<AccentIcon name={AccentIcon.Icons.Shield} color={colors.pomegranate500} />
<AccentIcon name={AccentIcon.Icons.Shield} size={80} />
```

#### Properties

**Required**

**`name`**  
Name of the icon. Must be a member of the `AccentIcon.Icons` static enum.

**Optional**

**`color`**  
Default: `colors.blueberry500`  
Icon color.

**`size`**  
Default: `56`  
Icon size in `px`.

### Applet Icons

```jsx
import { AppletIcon } from '@workday/canvas-kit-react-icon'

<AppletIcon name={AppletIcon.Icons.Benefits} />
<AppletIcon name={AppletIcon.Icons.Benefits} color={AppletIcon.Colors.Pomegranate} />
<AppletIcon name={AppletIcon.Icons.Benefits} size={60} />
```

#### Properties

**Required**

**`name`**  
Name of the icon. Must be a member of the `AppletIcon.Icons` static enum.

**Optional**

**`color`**  
Default: `AppletIcon.Colors.Blueberry`  
Icon color. Must use a member of the `AppletIcon.Colors` enum.

**`size`**  
Default: `92`  
Icon size in `px`.

### System Icons

```jsx
import { colors } from '@workday/canvas-kit-react-core'
import { SystemIcon } from '@workday/canvas-kit-react-icon'

<SystemIcon name={SystemIcon.Icons.ActivityStream} />
<SystemIcon name={SystemIcon.Icons.ActivityStream} color={colors.blueberry500} />
<SystemIcon
  name={SystemIcon.Icons.ActivityStream}
  accent={colors.frenchVanilla100}
  fill={colors.blueberry500}
  background={colors.blueberry500}
/>
<SystemIcon name={SystemIcon.Icons.ActivityStream} size={48} />
```

#### Properties

**Required**

**`name`**  
Name of the icon. Must be a member of the `SystemIcon.Icons` static enum.

**Optional**

**`size`**  
Default: `24`  
Icon size in `px`.

**`color`**  
Default: `colors.primary.iconStandard`  
Icon color. This will define `accent` and `fill`. `accent` and `fill` will override this property if defined.

**`colorHover`**  
Default: `colors.primary.iconHover`  
Icon hover color. This will define `accentHover` and `fillHover`. `accentHover` and `fillHover` will override this property if defined.

**`accent`**  
`.wd-icon-accent` color.

**`accentHover`**  
`.wd-icon-accent` color on hover.

**`background`**  
Default: `'transparent'`  
`.wd-icon-background` color.

**`backgroundHover`**  
Default: `'transparent'`  
`.wd-icon-background` color on hover.

**`fill`**  
`.wd-icon-fill` color.

**`fillHover`**  
`.wd-icon-fill` color on hover.
