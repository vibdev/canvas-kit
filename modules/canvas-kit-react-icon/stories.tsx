import React from 'react'
import { storiesOf } from '@storybook/react'
import withReadme from 'storybook-readme/with-readme'
import { colors } from '@workday/canvas-kit-react-core'

import Icon, { AccentIcons, AppletIcons, SystemIcons, IconSets } from './index' // tslint:disable-line:import-name
import README from './README.md'

storiesOf('Canvas Kit/Icon', module)
  .addDecorator(withReadme(README))
  .add('All', () => (
    <div className="story">
      <h1 className="section-label">Icon</h1>
      <h2>Accent Icon</h2>
      <Icon name={AccentIcons.Shield} set={IconSets.Accent} />
      <Icon name={AccentIcons.Shield} set={IconSets.Accent} color={colors.pomegranate500} />

      <h2>Applet Icon</h2>
      <Icon name={AppletIcons.Benefits} set={IconSets.Applet} />

      <h2>System Icon</h2>
      <Icon name={SystemIcons.ActivityStream} />
      <Icon name={SystemIcons.ActivityStream} color={colors.blueberry500} />
      <Icon
        name={SystemIcons.ActivityStream}
        accent={colors.frenchVanilla100}
        fill={colors.blueberry500}
        background={colors.blueberry500}
      />
    </div>
  ))
