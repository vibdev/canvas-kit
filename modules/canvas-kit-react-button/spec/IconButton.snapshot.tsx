import * as renderer from 'react-test-renderer';
import IconButton from '../lib/IconButton';
import * as React from 'react';
import {activityStreamIcon} from '@workday/canvas-system-icons-web';
import {SystemIcon} from '@workday/canvas-kit-react-icon';

describe('Icon Button Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(
      <IconButton>
        <svg>Hello</svg>
      </IconButton>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders an icon button, toggled', () => {
    const component = renderer.create(
      <IconButton toggled={true}>
        <SystemIcon icon={activityStreamIcon} />
      </IconButton>
    );
    expect(component).toMatchSnapshot();
  });
});
