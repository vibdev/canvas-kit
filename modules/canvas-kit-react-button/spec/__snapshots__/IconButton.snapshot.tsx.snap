// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Button Snapshots renders as expected 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  width: 40px;
  height: 40px;
  border-radius: 3px;
  border-width: 1px;
  width: 32px;
  height: 32px;
  background-color: #f0f1f2;
  border-color: #ced3d9;
  color: #333333;
}

.emotion-0:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-0:disabled,
.emotion-0:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-0 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-0 span .wd-icon-fill,
.emotion-0 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-0:focus {
  background-color: #f0f1f2;
  border-color: #005cb9;
}

.emotion-0:focus span .wd-icon-fill,
.emotion-0:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover:focus {
  background-color: #dfe2e6;
}

.emotion-0:active,
.emotion-0:focus:active,
.emotion-0:hover:active {
  background-color: #ced3d9;
}

.emotion-0:active span .wd-icon-fill,
.emotion-0:focus:active span .wd-icon-fill,
.emotion-0:hover:active span .wd-icon-fill,
.emotion-0:active span .wd-icon-accent,
.emotion-0:focus:active span .wd-icon-accent,
.emotion-0:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-0:hover {
  background-color: #dfe2e6;
}

.emotion-0:hover span .wd-icon-fill,
.emotion-0:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-0:disabled,
.emotion-0:active:disabled,
.emotion-0:focus:disabled,
.emotion-0:hover:disabled {
  background-color: #ffffff;
}

.emotion-0:disabled span .wd-icon-fill,
.emotion-0:active:disabled span .wd-icon-fill,
.emotion-0:focus:disabled span .wd-icon-fill,
.emotion-0:hover:disabled span .wd-icon-fill,
.emotion-0:disabled span .wd-icon-accent,
.emotion-0:active:disabled span .wd-icon-accent,
.emotion-0:focus:disabled span .wd-icon-accent,
.emotion-0:hover:disabled span .wd-icon-accent {
  fill: #f0f1f2;
}

.emotion-0:not([disabled]):focus,
.emotion-0:not([disabled]):active {
  border-color: #005cb9;
  -webkit-animation: animation-1lvxwzr 100ms;
  animation: animation-1lvxwzr 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 1px #005cb9;
}

<button
  className="emotion-0"
>
  <svg>
    Hello
  </svg>
</button>
`;
